src/compiler/checker.ts

```diff
@@ -14133,19 +14178,100 @@ namespace ts {
                     }
                 }
 
-                // Checks for ambient external modules.
                 if (isAmbientExternalModule) {
-                    if (!isGlobalSourceFile(node.parent)) {
-                        error(node.name, Diagnostics.Ambient_modules_cannot_be_nested_in_other_modules_or_namespaces);
+                    if (isExternalModuleAugmentation(node)) {
+                        // if symbol of augmentation is not merged this means that either
+                        // - this is an augmentation of the global scope 
+                        // or
+                        // - this augmentation was not merged with main definition of the module
+                        //   error should already be reported so all errors in the body of augmentation can be ignored.
+                        const checkBody = isNameOfGlobalAugmentation(<LiteralExpression>node.name) || (getSymbolOfNode(node).flags & SymbolFlags.Merged);
+                        if (checkBody) {
+                            const globalAugmentation = isNameOfGlobalAugmentation(<LiteralExpression>node.name);
+                            // body of ambient external module is always a module block
+                            for (const statement of (<ModuleBlock>node.body).statements) {
+                                checkBodyOfModuleAugmentation(statement, globalAugmentation);
+                            }
+                        }
                     }
-                    if (isExternalModuleNameRelative(node.name.text)) {
-                        error(node.name, Diagnostics.Ambient_module_declaration_cannot_specify_relative_module_name);
+                    else if (isGlobalSourceFile(node.parent)) {
+                        if (isExternalModuleNameRelative(node.name.text)) {
+                            error(node.name, Diagnostics.Ambient_module_declaration_cannot_specify_relative_module_name);

```

Comment:
This should be `An ambient module declaration cannot specify a relative module name.`